web/frontend/src/lib/components/robot-client.svelte

```diff
@@ -196,71 +201,75 @@ const fetchCurrentSessions = async () => {
   for (const grpcStatus of grpcStatuses) {
-    const nameObj = grpcStatus.getName()!.toObject();
-    const status = grpcStatus.getStatus()!.toJavaScript();
+    const nameObj = grpcStatus.name!;
+    const status = grpcStatus.status!.toJson();
     const name = resourceNameToString(nameObj);
 
     $statuses[name] = status;
   }
 };
 
 const restartStatusStream = () => {
-  if ($statusStream) {
-    $statusStream.cancel();
+  if ($statusStreamAbort) {
+    $statusStreamAbort.abort();
+    $statusStreamAbort = null;
     $statusStream = null;
   }
 
-  let newResources: commonApi.ResourceName.AsObject[] = [];
+  let newResources: ResourceName[] = [];
 
   // get all relevant resources
   for (const subtype of relevantSubtypesForStatus) {
     newResources = [...newResources, ...filterSubtype($components, subtype)];
   }
 
   const names = newResources.map((name) => {
-    const resourceName = new commonApi.ResourceName();
-    resourceName.setNamespace(name.namespace);
-    resourceName.setType(name.type);
-    resourceName.setSubtype(name.subtype);
-    resourceName.setName(name.name);
-    return resourceName;
+    return new ResourceName({
+      namespace: name.namespace,
+      type: name.type,
+      subtype: name.subtype,
+      name: name.name,
+    });
+  });
+
+  const streamReq = new robotApi.StreamStatusRequest({
+    resourceNames: names,
+    every: new Duration({
+      nanos: 500_000_000,
+    }),
   });
 
-  const streamReq = new robotApi.StreamStatusRequest();
-  streamReq.setResourceNamesList(names);
-  streamReq.setEvery(new Duration().setNanos(500_000_000));
+  $statusStreamAbort = new AbortController();
+  $statusStream = $robotClient.robotService.streamStatus(streamReq, {
+    signal: $statusStreamAbort.signal,
+  });
 
-  $statusStream = $robotClient.robotService.streamStatus(streamReq);
-  $statusStream.on(
-    'data',
-    (response: { getStatusList(): robotApi.Status[] }) => {
-      updateStatus(response.getStatusList());
-      lastStatusTS = Date.now();
-    }
-  );
-  $statusStream.on('status', (newStatus?: { details: unknown }) => {
-    if (!ConnectionClosedError.isError(newStatus!.details)) {
+  (async () => {
+    try {
+      for await (const resp of $statusStream) {
+        updateStatus(resp.status);
+        lastStatusTS = Date.now();
+      }
       // eslint-disable-next-line no-console
-      console.error('error streaming robot status', newStatus);
+      console.error('done streaming robot status');
+    } catch (error) {
+      if (!ConnectionClosedError.isError(error)) {
+        // eslint-disable-next-line no-console
+        console.error('error streaming robot status', error);
+      }
+    } finally {
+      $statusStream = null;
     }
-    $statusStream = null;
-  });
-  $statusStream.on('end', () => {
-    // eslint-disable-next-line no-console
-    console.error('done streaming robot status');
-    $statusStream = null;
-  });
+  })().catch(console.error);
```

Comment:
Will this catch actually catch anything since the whole thing is wrapped in a try/catch?

Reply:
nope good... catch